ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/device/tilize_with_val_padding_op.cpp

```diff
@@ -17,21 +17,23 @@ void TilizeWithValPadding::validate(const std::vector<Tensor>& input_tensors) co
     TT_FATAL(input_tensor_a.get_layout() == Layout::ROW_MAJOR, "Can only tilize row major data");
     TT_FATAL(input_tensor_a.get_dtype() == DataType::BFLOAT16);
 
-    TT_FATAL(input_tensor_a.get_legacy_shape()[0] <= this->output_tensor_shape[0]);
-    TT_FATAL(input_tensor_a.get_legacy_shape()[1] <= this->output_tensor_shape[1]);
-    TT_FATAL(input_tensor_a.get_legacy_shape()[2] <= this->output_tensor_shape[2]);
-    TT_FATAL(input_tensor_a.get_legacy_shape()[3] <= this->output_tensor_shape[3]);
+    const auto& input_shape = input_tensor_a.get_legacy_shape();
+    for (auto i = 0; i < input_shape.rank(); i++) {
+        TT_FATAL(input_tensor_a.get_legacy_shape()[i] <= this->output_tensor_shape[i],
+                 "Output shape {} must be greater than or equal to input shape {} in each dimension",
+                 this->output_tensor_shape, input_shape);
+    }
 
-    uint32_t num_rows = this->output_tensor_shape[2];
-    uint32_t inner_dim = this->output_tensor_shape[3];
-    TT_FATAL(num_rows % TILE_HEIGHT == 0, "Output shape must be tilizable");
-    TT_FATAL(inner_dim % TILE_WIDTH == 0, "Output shape must be tilizable");
+    uint32_t num_rows = this->output_tensor_shape[input_shape.rank() - 1];
+    uint32_t inner_dim = this->output_tensor_shape[input_shape.rank() - 2];
+    TT_FATAL(num_rows % TILE_HEIGHT == 0, "Output shape must be tilizable, but {} is not divisible by {}", num_rows, TILE_HEIGHT);
+    TT_FATAL(inner_dim % TILE_WIDTH == 0, "Output shape must be tilizable, but {} is not divisible by {}", inner_dim, TILE_WIDTH);
 
     if (input_tensor_a.memory_config().is_sharded()) {
         TT_FATAL(input_tensor_a.memory_config().memory_layout == TensorMemoryLayout::WIDTH_SHARDED);
         TT_FATAL(this->output_mem_config.memory_layout == input_tensor_a.memory_config().memory_layout);
         for (uint32_t i = 0; i < input_tensor_a.get_legacy_shape().rank(); i++) {
```

Comment:
This loop is really weird. I am not sure what author wanted to check with it.
I guess maybe the intension was to compare that non-padded dims (everything except last 2 dims) are the same in input and output. 
But then unclear why this check is only inside "is sharded" branch and why the above for loop checks the whole shape...